lib/actions/appleMusic.js

```diff
@@ -155,15 +145,16 @@ function appleMusic(player, values) {
                 return player.coordinator.clearQueue()
                   .then(() => player.coordinator.addURIToQueue(queueTracks[0].uri, queueTracks[0].metadata, true, 1))
                   .then(() => {
-                      if (radioOn(player)) {
-                        return promise.then(() => player.coordinator.setAVTransport(queueURI,''));
+                      if (isRadioOrLineIn(player.coordinator.state.currentTrack.uri)) {
+                        player.coordinator.setAVTransport(queueURI,''));
                       }
+                      return Promise.resolve();
                     })
                   .then(() => player.coordinator.play())
                   .then(() => {
                     return queueTracks.reduce(function(promise, track, index){
                       if (index > 0) {
-                        return promise.then(() => player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1));
+                        player.coordinator.addURIToQueue(track.uri, track.metadata, true, index + 1);
                       } else {
                         return promise.then(() => {});
                       }   
```

Comment:
This is probably the culprit for the quick exit, since you aren't creating a promise chain here, you are just starting a lot of promises. 

If you'd like, I could fix this according to your liking (return after first track has been queued and started and then continue adding tracks, for example).